isis/src/system/apps/kerneldbgen/SpiceDbGen.cpp

```diff
@@ -131,6 +130,79 @@ PvlObject SpiceDbGen::Direct(QString quality, QString location,
   return result;
 }
 
+
+/**
+ * Creates a Pvl object that stores all of the kernels specified by a list
+ *
+ * @param quality   The quality of the kernels that are being filtered into
+ *                  the database. For example, "Reconstructed".
+ *
+ * @param fileList  The list of files to create a database for. The files must
+ *                  be ordered in ascending priority. That is, if two kernels
+ *                  cover the same time period, the kernel later in the file
+ *                  will be used.
+ *
+ * @return PvlObject
+ *
+ * @internal
+ *   @history 2019-08-13 Jesse Mapel Created method that takes a file list.
+ *
+ * @throws Isis::iException::Message
+*/
+PvlObject SpiceDbGen::Direct(QString quality, FileList fileList,
+                             double startOffset, double endOffset) {
+  PvlObject result;
+  QString type = "none";
+
+  // Throw an error if no files are being added to this database for
+  // this filter/regex
+  if (fileList.empty()) {
+    QString message = "Input filelist is empty!";
+    throw IException(IException::User, message, _FILEINFO_);
+  }
+
+  for (int fileNum = 0 ; fileNum < fileList.size() ; fileNum++) {
+    FileName currFile = fileList[fileNum];
+    PvlGroup selection = AddSelection(currFile, startOffset, endOffset);
+    selection += PvlKeyword("Type", quality);
+    result.addGroup(selection);
+  }
+
+  // Check each group to make sure it is the same type as others
+  PvlObject::PvlGroupIterator grp = result.beginGroup();
+
+  while (grp != result.endGroup()) {
+    // The kernel did not have any time coverage, so ignore it
+    if (grp->name() == "No coverage" || grp->name() == "Null") {
+      result.deleteGroup(grp->name());
+    }
+    // This is used for the first time thru the while loop
+    // DO NOT increment grp here
+    else if (type == "none") {
+      type = grp->name();
+    }
+    else if (grp->name() == type) {
+      grp->setName("Selection");
+      grp++;
+    }
+    else {
+      QString message = "A kernel of type [" + grp->name() + "] has been found in a directory for type [" + type + "]" ;
+      throw IException(IException::Programmer, message, _FILEINFO_);
+      break;
+    }
```

Comment:
I just want to make sure that I am understanding this if/else block: is this assuming that the first file of valid time range is the type that was intended? Is there ever, for example, a time when the first file is a fluke somehow (with an "SPK" type), but all others have a "CK" type (and the actual type is meant to be "CK")? If so, this error message may be misleading...?

Reply:
Yeah this is some weird logic. It should just check that the name is equal to the type for each group.